website/docs/cloud-docs/agents/agents.mdx

```diff
@@ -85,7 +85,11 @@ An image customized in this way permits installation of additional software via
 
 ### Optional Configuration: Single-Execution Mode
 
-You can also configure the agent to run in single-execution mode, which ensures that the agent only runs a single workload, then terminates. You can use this configuration in combination with Docker and a process supervisor to ensure a clean working environment for every Terraform run.
+You can also configure the agent to run in single-execution mode, which ensures that the agent only runs a single workload, then terminates. You can use this configuration in combination with Docker and a process supervisor to ensure a clean working environment for every Terraform run. Agents in single -execution mode perform one operation at a time.  Plan and apply are separate operations and would require an agent per operation.  
+
+~>**Important:** When an agent is placed in Single-Execution Mode, if the plan and apply operation is queued from the UI, after the plan has completed its part the agent (placed in single-execution mode) will exit, leaving the apply operation without a agent to process against.  The apply operation will then enter in a queued state until an agent process is available.  This is the expected behavior as agents in single mode perform one operation at a time. 
+
+~>To get around this you’d need to create two distinct agents in single execution mode, under the same agent pool that is configured for the associated workspace execution mode to work through a whole queue plan and apply run operation. 
```

Comment:
```suggestion
You can also configure the agent to run in single-execution mode, which ensures that the agent only runs a single workload, then terminates. You can use this configuration in combination with Docker and a process supervisor to ensure a clean working environment for every Terraform run. 

Agents in single-execution mode perform one operation at a time. Plan and apply are separate operations, so they each require an agent. When you queue a plan and apply from the Terraform Cloud UI, agents in single-execution mode exit after the plan is complete. The apply operation then remains in a queued state until an agent process is available. 

To change this behavior, create two agents in single-execution mode and add them to the same agent pool. Then, you configure the agent pool to handle the entire plan and apply run within workspaces. 
```

Reply:
I think this suggestion needs to be merged in prior to merging the PR 👌